local_planner/test/test_common.cpp

```diff
@@ -26,3 +34,369 @@ TEST(Common, polar2DdistanceOnKnownPoints) {
+
+  //THEN: the vertical histogram index should be ..
+  EXPECT_EQ(30, index_1);
+  EXPECT_EQ(41, index_2);
+  EXPECT_EQ(7, index_3);
+  EXPECT_EQ(10, index_4);
+  EXPECT_EQ(14, index_5);
+  EXPECT_EQ(0, index_6);
+  EXPECT_EQ(0, index_invalid_1);
+  EXPECT_EQ(0, index_invalid_2);
+  EXPECT_EQ(0, index_invalid_3);
+  EXPECT_EQ(0, index_invalid_4);
+
+}
+TEST(Common, azimuthAngletoIndex){
+  //GIVEN: the azimuth angle of a point and the histogram resolution
+  const float elevation_1 = 0.f;
+  const float elevation_2 = 34.f;
+  const float elevation_3 = 180.f;
+  const float elevation_4 = 179.f;
+  const float elevation_5 = -180.f;
+  const float resolution_1 = 3.f;
+  const float resolution_2 = 12.f;
+  const float elevation_invalid_1 = 194.f;
+  const float elevation_invalid_2 = -999.f;
+  const float resolution_invalid_1 = 0.f;
+  const float resolution_invalid_2 = -1.f;
+
+  //WHEN: we convert the azimuth angle to a histogram index
+  const int index_1 = azimuthAngletoIndex(elevation_1, resolution_1);
+  const int index_2 = azimuthAngletoIndex(elevation_2, resolution_1);
+  const int index_3 = azimuthAngletoIndex(elevation_1, resolution_2);
+  const int index_4 = azimuthAngletoIndex(elevation_2, resolution_2);
+  const int index_5 = azimuthAngletoIndex(elevation_3, resolution_2);
+  const int index_6 = azimuthAngletoIndex(elevation_4, resolution_2);
+  const int index_7 = azimuthAngletoIndex(elevation_5, resolution_2);
+  const int index_invalid_1 = azimuthAngletoIndex(elevation_invalid_1, resolution_1);
+  const int index_invalid_2 = azimuthAngletoIndex(elevation_invalid_2, resolution_1);
+  const int index_invalid_3 = azimuthAngletoIndex(elevation_1, resolution_invalid_1);
+  const int index_invalid_4 = azimuthAngletoIndex(elevation_1, resolution_invalid_2);
+
+  //THEN: the horizontal histogram index should be ..
+  EXPECT_EQ(60, index_1);
+  EXPECT_EQ(71, index_2);
+  EXPECT_EQ(15, index_3);
+  EXPECT_EQ(17, index_4);
+  EXPECT_EQ(0, index_5);
+  EXPECT_EQ(29, index_6);
+  EXPECT_EQ(0, index_7);
+  EXPECT_EQ(0, index_invalid_1);
+  EXPECT_EQ(0, index_invalid_2);
+  EXPECT_EQ(0, index_invalid_3);
+  EXPECT_EQ(0, index_invalid_4);
+
+}
+
+TEST(Common, fromPolarToCartesian){
+
+//GIVEN: the elevation angle, azimuth angle, a radius and the position
+std::vector<float> e = {-90.f, -90.f, 90.f, 0.f, 45.f};  //[-90, 90]
+std::vector<float> z = {-180.f, -90.f, 179.f, 0.f, 45.f}; //[-180, 180]
+
+std::vector<double> radius = {0.f, 2.f};
+
+geometry_msgs::Point pos;
+pos.x = 0.f;
+pos.y = 0.f;
+pos.z = 0.f;
+
+std::vector<geometry_msgs::Point> pos_out;
+
+//WHEN: converting the point in polar CS to cartesian CS
+
+for(int i = 0; i<e.size(); i++){
+     pos_out.push_back(fromPolarToCartesian(e[i], z[3], radius[0], pos));
+}
+
+for(int i = 0; i<e.size(); i++){  
+       pos_out.push_back(fromPolarToCartesian(e[i], z[i], radius[1], pos));
+}
+
+
+//THEN: the cartesian coordinates are
+
+for(int i = 0; i<e.size(); i++){
```

Comment:
Again, checking for the size of `e` when actually iterating over some other vector is not very safe. Consider something like `for (auto i : pos_out){`